internal/querynode/collection.go

```diff
@@ -170,15 +185,21 @@ OUTER:
 
 // getPChannels get physical channels of collection
 func (c *Collection) getPDeltaChannels() []Channel {
+	c.channelMu.RLock()
+	defer c.channelMu.RUnlock()
 	return c.pDeltaChannels
 }
 
 func (c *Collection) getVDeltaChannels() []Channel {
+	c.channelMu.RLock()
+	defer c.channelMu.RUnlock()
 	return c.vDeltaChannels
```

Comment:
same here, should do copy before release

Reply:
fixed